pkg/test/framework/components/istio/operator.go

```diff
@@ -366,12 +367,31 @@ func deploy(ctx resource.Context, env *kube.Environment, cfg Config) (Instance,
 	}
 
 	// Install (non-config) remote clusters.
+	// Need to determine if there is a setting to watch cluster secret in config cluster
+	// or in external cluster. The flag is named LOCAL_CLUSTER_SECERT_WATCHER and set as
+	// an environment variable for istiod.
+	var localClusterSecretWatcher *structpb.Value
+	if istioctlConfigFiles.operatorSpec != nil && istioctlConfigFiles.operatorSpec.Values != nil {
+		localClusterSecretWatcher = GetConfigValue("pilot.env.LOCAL_CLUSTER_SECERT_WATCHER",
+			istioctlConfigFiles.operatorSpec.Values.Fields)
+	} else {
+		localClusterSecretWatcher = structpb.NewNullValue()
+	}
 	errG = multierror.Group{}
 	for _, c := range ctx.Clusters().Kube().Remotes(ctx.Clusters().Configs()...) {
 		c := c
 		errG.Go(func() error {
-			// Configure API server access for the remote cluster's primary cluster control plane.
-			if err := i.configureDirectAPIServiceAccessBetweenClusters(ctx, cfg, c, c.Config()); err != nil {
+			// Configure API server access for the remote cluster's primary cluster control plane
+			// which can be either config cluster or external cluster
+			var from []cluster.Cluster
+			if localClusterSecretWatcher.GetStringValue() == "true" {
+				// target clusters will be external cluster
+				from = append(from, ctx.AllClusters().Kube().Primaries(c.Config())...)
```

Comment:
What is this doing? Why not just c.Primary()?

Reply:
this to get remote secrets to external (non-config) clusters.

Reply:
c.Primary()